magefiles/installation/oauth.go

```diff
@@ -0,0 +1,120 @@
+package installation
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	retry "github.com/avast/retry-go/v4"
+	"github.com/devfile/library/pkg/util"
+	ocpOauth "github.com/openshift/api/config/v1"
+	"github.com/redhat-appstudio/e2e-tests/pkg/utils"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	HTPASSWD_FILE_NAME      = "ci.htpaswd"
+	DEFAULT_IDP_SECRET_NAME = "htpass-secret"
+	DEFAULT_IDP_NAME        = "ci_identity"
+	DEFAULT_OAUTH_NAME      = "cluster"
+	DEFAULT_HTPASSWD_BIN    = "htpasswd"
+)
+
+var (
+	randomOCPUserPass = util.GenerateRandomString(10)
+	randomOCPUserName = util.GenerateRandomString(6)
+	httpaswdArgs      = []string{"-c", "-B", "-b", HTPASSWD_FILE_NAME, randomOCPUserName, randomOCPUserPass}
+)
+
+// CreateOauth generate a new random admin user for testing. This functionality will be executed only in Openshift CI.
+func (i *InstallAppStudio) CreateOauth() error {
+	if os.Getenv("CI") != "true" {
+		return nil
+	}
+
+	if err := i.generateHttpaswd(); err != nil {
+		return err
+	}
+
+	if err := utils.ExecuteCommandInASpecificDirectory("oc", []string{"create", "secret", "generic", DEFAULT_IDP_SECRET_NAME, fmt.Sprintf("--from-file=htpasswd=%s/%s", i.TmpDirectory, HTPASSWD_FILE_NAME), "-n", "openshift-config"}, ""); err != nil {
```

Comment:
can we use golang library to create this secret?

Reply:
Make sense in this case!